fauna/query_builder.py

```diff
@@ -63,39 +57,18 @@ class LiteralFragment(Fragment):
     def __init__(self, val: str):
         self._val = val
 
-    def render(self) -> str:
-        """Renders the :class:`LiteralFragment` into the wire protocol for a literal of the query template API.
-
-        e.g. ``let x = ``
+    def get(self) -> str:
+        """Returns the stored value.
 
-        :returns: The query literal.
+        :returns: The stored value.
         """
         return self._val
 
 
-class QueryFragment(Fragment):
-    """A concrete :class:`Fragment` representing a subquery within a template. For example, if a template contains a
-    variable ```${foo_query}```, and a :class:`QueryBuilder` is provided for foo, then it should be wrapped as a
-    :class:`QueryFragment`.
+class QueryInterpolationBuilder(QueryBuilder):
```

Comment:
Do we even need "builders" now? Can this still be a `QueryFragment` whose `_val` is a `List[Fragment]`?

Other builder method have been taken away. What's left is `fragments` which a simple getter. At this point the `QueryBuilder` class looks a lot like a `Fragment` sub class.

Reply:
The builder is the contract the for the query API, which is different from a fragment.

For instance, if we move this to be just a Fragment, then the API either needs to accept a QueryFragment or a Fragment. The latter may not always make sense (i.e. a Fragment isn't by definition always a valid query. The former, for it to be extensible, would have to be abstract and then we'd implement QueryInteroplationFragment.

I think keeping these ideas separate means we can have an extensible yet safe API contract and we don't run the risk of Fragments and their uses being in conflict with a query builder and their uses.

Reply:
We could rename these if you think that'd be an improvement, though. Like `QueryInteropolation` and an abstract `Query`

Reply:
Taking this a bit further, maybe removing the abstract class altogether for now and treat QueryInterpolation as a collection of fragments. Then another query type doesn't necessarily need to rely on fragments.